### ISSUE TITLE
CI fails with JavaScript heap out of memory error

### ISSUE THREAD

**Issue by ventuno** [ember-template-lint/ember-template-lint#2599]
Currently open PRs are failing some of the CI jobs with the following error:
```
<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb6e260 node::Abort() [node]
 2: 0xa7cdea  [node]
 3: 0xd45f00 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xd462a7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xf235b5  [node]
 6: 0xf244b8 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [node]
 7: 0xf349c3  [node]
 8: 0xf35838 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 9: 0xf1019e v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
10: 0xf11567 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
11: 0xef273a v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
12: 0x12b5c4f v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
13: 0x16e78b9  [node]
Error: Process completed with exit code 1.
```

Examples:
* https://github.com/ember-template-lint/ember-template-lint/pull/2483
* https://github.com/ember-template-lint/ember-template-lint/pull/2593
* https://github.com/ember-template-lint/ember-template-lint/pull/2597

**Comment by bmish** [comment]
Yep this is annoying, I have been ignoring the failure and merging PRs still, would appreciate any fix to this.

**Comment by nlfurniss** [comment]
Is this reproducible locally?

**Comment by ventuno** [comment]
@nlfurniss not for me. I could run `yarn run test` without incurring in this error.

**Comment by nlfurniss** [comment]
Changed node to be `18.80` (same as installed on CI), ran the tests, and got this:

```sh
Summary of all failing tests
 FAIL  test/acceptance/todo-cli-test.js (150.47 s)
  ● todo usage › with/without --update-todo and --include-todo params › cleaning todos in CI › errors if a todo item is no longer valid when running without params, and cleans using --fix

    thrown: "Exceeded timeout of 10000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      304 |       setupEnvVar('GITHUB_ACTIONS', true);
      305 |
    > 306 |       it('errors if a todo item is no longer valid when running without params, and cleans using --fix', async function () {
          |       ^
      307 |         await project.setConfig({
      308 |           rules: {
      309 |             'require-button-type': true,

      at it (test/acceptance/todo-cli-test.js:306:7)
      at describe (test/acceptance/todo-cli-test.js:302:5)
      at describe (test/acceptance/todo-cli-test.js:28:3)
      at describe (test/acceptance/todo-cli-test.js:14:1)


Test Suites: 1 failed, 140 passed, 141 total
Tests:       1 failed, 5 skipped, 7863 passed, 7869 total
Snapshots:   1028 passed, 1028 total
Time:        151.975 s
Ran all test suites.
```

Sadly no memory issues 🤔

**Comment by nlfurniss** [comment]
This "fixes" the issue:

```diff
// test/acceptance/todo-cli-test.js
-jest.setTimeout(10_000);
+jest.setTimeout(100_000);
```

**Comment by rwjblue** [comment]
"Fixed" by #2661
